packages/langium/src/workspace/index-manager.ts

```diff
@@ -151,6 +161,17 @@ export class DefaultIndexManager implements IndexManager {
         this.referenceIndex.delete(uriString);
     }
 
+    removeContent(uri: URI): void {
+        const uriString = uri.toString();
+        this.symbolIndex.delete(uriString);
+        this.symbolByTypeIndex.clear(uriString);
+    }
+
+    removeReferences(uri: URI): void {
+        const uriString = uri.toString();
+        this.referenceIndex.delete(uriString);
+    }
```

Comment:
Suggestion: How about making `remove` simply call both `removeContent` and `removeReferences`?

Reply:
👍 I thought that would stringify the URI twice, but after a deeper look I saw that URI caches its string rep, and we call this only when deleting documents.